src/lib/types/color16.js

```diff
@@ -0,0 +1,15 @@
+import r from 'restructure';
+
+class Color16 {
```

Comment:
It'd probably be okay to call this something like ```SignedFixed16``` and read from the stream as an ```int16le``` rather than ```uint16le```, but I find it weird to do that for values that can't be negative (like transparency and color values). 

Hence... ```Color16``` -- your one-stop shop for values between 0.0 and 1.0 stored as shorts with values between 0 and 32767.